djangocms_picture/models.py

```diff
@@ -1,83 +1,236 @@
+    use_no_cropping = models.BooleanField(
+        verbose_name=_('Use original image.'),
+        blank=True,
+        default=False,
+        help_text=_('Outputs the raw image without cropping.'),
+    )
+    # upscale and crop work together
+    # throws validation error if other cropping options are selected
+    use_crop = models.BooleanField(
+        verbose_name=_('Crop image'),
+        blank=True,
+        default=False,
+        help_text=_('Crops the image according to the given thumbnail settings in the template.'),
+    )
+    use_upscale = models.BooleanField(
+        verbose_name=_('Upscale image'),
+        blank=True,
+        default=False,
+        help_text=_('Upscales the image to the size of the thumbnail settings in the template.')
+    )
+    # overrides all other options
+    # throws validation error if other cropping options are selected
+    use_thumbnail = models.ForeignKey(
+        ThumbnailOption,
+        verbose_name=_('Thumbnail options'),
+        blank=True,
+        null=True,
+        help_text=_('Overrides width, height, crop and upscale with the provided preset.'),
+    )
 
-    page_link = models.ForeignKey(
-        Page, verbose_name=_("page"), null=True,
-        limit_choices_to={'publisher_is_draft': True}, blank=True,
-        help_text=_("If present, clicking on image will take user to "
-                    "specified page."))
+    # Add an app namespace to related_name to avoid field name clashes
+    # with any other plugins that have a field with the same name as the
+    # lowercase of the class name of this model.
+    # https://github.com/divio/django-cms/issues/5030
+    cmsplugin_ptr = models.OneToOneField(
+        CMSPlugin,
+        related_name='%(app_label)s_%(class)s',
+        parent_link=True,
+    )
 
-    alt = models.CharField(
-        _("alternate text"), max_length=255, blank=True, null=True,
-        help_text=_("Specifies an alternate text for an image, if the image"
-                    "cannot be displayed.<br />Is also used by search engines"
-                    "to classify the image."))
+    def __str__(self):
+        if self.picture and self.picture.label:
+            return self.picture.label
+        if self.link_url:
+            return self.link_url
+        return ugettext('<file is missing>')
 
-    longdesc = models.CharField(
-        _("long description"), max_length=255, blank=True, null=True,
-        help_text=_("When user hovers above picture, this text will appear "
-                    "in a popup."))
+    def get_size(self):
+        # automatic scaling
+        # TODO need to get the base values
+        width = 0
+        height = 0
+        crop = False
+        upscale = False
 
-    float = models.CharField(
-        _("side"), max_length=10, blank=True, null=True, choices=FLOAT_CHOICES,
-        help_text=_("Move image left, right or center."))
+        # use width and height attributes
+        if self.width:
```

Comment:
aye thats right, though how can I get the placeholder dimensions in there? Like https://github.com/divio/cmsplugin-filer/blob/develop/cmsplugin_filer_image/cms_plugins.py#L71